pkg/util/machines.go

```diff
@@ -109,10 +109,21 @@ func GetMachineMetadata(hostname string, vsphereVM infrav1.VSphereVM, ipamState
 			devices[i].Gateway6 = state.Gateway6
 		}
 
+		// Ignore device if SkipIPAllocation is set
```

Comment:
Can we add why we want to skip this device here? What impact has that?

Reply:
After some consideration I think it's not really necessary. If the device has no static IP address or DHCP configured, then `waitForIPv4` and `waitForIPv6` should be `false` anyway.